dashboard/src/proxy.ts

```diff
@@ -200,13 +72,40 @@ export class ProxyDashboard {
     // 2. Disable inspection if user moved away from inspect tab
     // 3. Do nothing if activeTabId == clickedTabId
     if (clickedTabId !== activeTabId) {
-      if (clickedTabId === 'proxyInspect') {
-        this.websocketApi.enableInspection()
-      } else if (activeTabId === 'proxyInspect') {
-        this.websocketApi.disableInspection()
+      //deactivateAll
+      this.home.disable();
+      this.apiDevelopment.disable();
+      this.websocketApi.disable();  
+      this.shortLinks.disable();
+      this.controls.disable();
+      this.settings.disable();
+
+      switch(clickedTabId) {
+        case "proxyHome": 
```

Comment:
Let's preserve this comment from previous version of file:

`// TODO: Tab ids shouldn't be hardcoded.`

Ideally, at some point, we want IDs in a single place so that refactoring becomes easy.  Keeping IDs within individual file is a good idea.